### DISCUSSION TITLE
(use case): Discourse & github-readme-stats

### DISCUSSION THREAD

**Discussion by ghost** [anuraghazra/github-readme-stats#2017] [Q&A] [answered]
Hi everyone! 

I would like to help this project and I have an idea that I think I could collaborate with. Maybe... this could make the repository more popular among many open source communities. Because most open source communities use things like Discourse and GitHub. 

So... this will allow to know the amount of answers, topics that open which is somehow similar to what I requested here: [1991](https://github.com/anuraghazra/github-readme-stats/issues/1991). In theory you would have to send and receive as a parameter the code of the [discourse-api-sdk](https://github.com/99x/discourse-sdk) as:

#### Data
![image](https://user-images.githubusercontent.com/111701513/189554684-fc6b9505-c905-4c19-82f2-c5c0cf12d23e.png)

#### Sample data
```json
{
 "days visited": "117 days visited",
 "read time": "13h read time",
 "recent read time":"3h recent read time",
 "topics viewed":"186 topics viewed",
 "posts read":"1.1k posts read",
 "given":"33 given",
 "received":"71 received",
 "topics created":"19 topics created",
 "posts created":"36 posts created"
}
```

#### Demo
![image](https://user-images.githubusercontent.com/111701513/189555166-d6e29fe9-95d8-4b5a-9c1a-c905fd80bff3.png)

#### Sample API - discourse_api.js
```discourse_api.js
var discourse = require('discourse-sdk');

function GetCategories(apiUrl, apiKey, username){
  let client = GetDiscourseByClient(apiUrl, apiKey, username)
  client.getCategories({},function(error, body, httpCode) {              
    return body;   // add count categories      
 });
}

function GetCategoryLatestTopic(apiUrl, apiKey, username){
  let client = GetDiscourseByClient(apiUrl, apiKey, username)
  client.getCategoryLatestTopic('category_slug', 'params',,function(error, body, httpCode) {              
   return body; // add count all latest topics   
  });
}

function GetCreatedTopics(apiUrl, apiKey, username){
   let client = GetDiscourseByClient(apiUrl, apiKey, username)
   client.getCreatedTopicsfunction('username' ,function(error, body, httpCode) {              
    return body;   // add count all topics created             
  });
}

function GetUserInformation(apiUrl, apiKey, username){
   let client = GetDiscourseByClient(apiUrl, apiKey, username)
   client.getUser(username, function(error, body, httpCode) {              
    return body;   // get username          
 });
}
```

#### model
```markdown
https://github-readme-stats.vercel.app/api?username=codehangen&discourse_apiUrl=apiUrl&discourse_apikey=apiKey&discourse_username=username&show_icons=true&theme_name='name'&title_color="2f80ed"&icon_color="4c71f2"&text_color="434d58"&bg_color="fffefe"&border_color="e4e2e2"
```

#### URL
```python
https://github-readme-stats.vercel.app/api?username=codehangen
&discourse_apiUrl="apiUrl"
&discourse_apikey="apiKey"
&discourse_username="username"
&show_icons=true
&theme_name='name'
&title_color="2f80ed"&
icon_color="4c71f2"&
text_color="434d58"&
bg_color="fffefe"&
border_color="e4e2e2"
```

#### URL and parameters
- ```discourse_apiUrl```: ```apiUrl```
- ```discourse_apiKey```: ```apiKey```
- ```discourse_username```: ```username```

**Comment by rickstaa** [comment, marked as answer]
@codehangen I like the idea. However, I think adding new cards is not desirable since the PAT load is already too high (see https://github.com/anuraghazra/github-readme-stats/issues/1563#issuecomment-1137509321). Because of that, I proposed to switch to an organisational structure (see #1867). Using an organization would allow us to better manage the loads between the different cards and enable people to create their own cards on top of the `github-readme-stats` package.

**Reply by ghost** [reply]
Hi rickstaa, thank you for feedback.
